lib/datasets.go

```diff
@@ -1723,3 +1730,67 @@ func formFileDataset(r *http.Request, ds *dataset.Dataset) (err error) {
 
 	return
 }
+
+// datasetImpl holds the method implementations for DatasetMethods
+type datasetImpl struct{}
+
+// List gets the reflist for either the local repo or a peer
+func (datasetImpl) List(scope scope, p *ListParams) ([]dsref.VersionInfo, error) {
+	return nil, fmt.Errorf("not yet implemented")
+}
+
```

Comment:
we need to have some implementation of each of the `DatasetMethods` on the `datasetImpl` struct, or the `buildMethodMap` we use when we build up the `Dispatch` sigfaults.

Reply:
This is a good pattern to put in place, it acts as a TODO list for methods that aren't switched to dispatch yet, and makes it easier to start converting functions one at a time.